added flatten plugin to remove parent from generated pom.xml in bom

lpandzic opened the pull request:
Flattening bom in order to prevent issues like https://github.com/querydsl/querydsl/issues/2921 or that is - to have a clear and simple bom.

lpandzic commented:
@jwgmeligmeyling can you review please?

F43nd1r commented on querydsl-bom/pom.xml:
```diff
@@ -16,6 +16,10 @@
     <packaging>pom</packaging>
 
+    <properties>
+        <flatten-maven-plugin.version>1.2.7</flatten-maven-plugin.version>
```
This property shows up in the generated pom. Is that necessary?

lpandzic replied:
Removed it from flattened pom.

F43nd1r reviewed (changes requested):
Please add the generated file to gitignore.

Also some modules are missing from bom (at least `guava` and `kotlin-codegen`), you can either address that here or I'll open another PR.

lpandzic commented:
> Please add the generated file to gitignore.
> 
> Also some modules are missing from bom (at least `guava` and `kotlin-codegen`), you can either address that here or I'll open another PR.

Added flattened-pom to gitignore.
Added missing modules.

lpandzic commented:
I don't have write access permission, can I get one or can someone merge the PR?

jwgmeligmeyling commented:
Thanks for looking into this! 👍🏻
